### ISSUE TITLE
training iterations

### ISSUE THREAD

**Issue by tgc1997** [r00tman/EventNeRF#1]
Nice work, I have a question about the training.
In your paper, you train the model for 5 * 10^5 iterations. However, N_iters in config.txt is set to 5000001. What is the correct number of parameters?

**Comment by circle-DING** [comment]
I have the same question.I modify parameters in configs/lego/test1_events.txt,changed N_iters (5000001 to 500001) and lrate_ Decay_ Steps (50000000 to 5000000), however, the image rendered based on readme is pure pink purple.Have you resolved this question？

**Comment by r00tman** [comment]
Hello, sorry for the delay.
The correct value is 5 * 10^5 iterations as in the paper. The value in the config (5*10^6) is irrelevant as we stopped training manually at 5 * 10^5 iterations.

**Comment by r00tman** [comment]
@circle-DING , could you please share the results?

**Comment by r00tman** [comment]
I'll share pretrained models today for easier reproduction

**Comment by circle-DING** [comment]
> @circle-DING , could you please share the results?

Due to limited computing resources, I'm sorry that I didn't train for 5*10^6 iterations. I stopped training at slightly over 5*10^5 iterations, which happens to be the correct approach you suggested.

**Comment by circle-DING** [comment]
> I'll share pretrained models today for easier reproduction

Thank you for your reply and help！

**Comment by 520jz** [comment]
> Hello, sorry for the delay. The correct value is 5 * 10^5 iterations as in the paper. The value in the config (5*10^6) is irrelevant as we stopped training manually at 5 * 10^5 iterations.

However, I found that if the number of iterations is set at 5x10^5 of the paper, the PSNR will not achieve the effect of the paper. I trained the lego scene with four 3090s, iterated 1 million times, and the PSNR was just 20, but in the paper it was 25

**Comment by 520jz** [comment]
> However, I found that if the number of iterations is set at 5x10^5 of the paper, the PSNR will not achieve the effect of the paper. I trained the lego scene with four 3090s, iterated 1 million times, and the PSNR was just 20, but in the paper it was 25

However, I found that if the number of iterations is set at 5x10^5 of the paper, the PSNR will not achieve the effect of the paper. I trained the lego scene with four 3090s, iterated 1 million times, and the PSNR was just 20, but in the paper it was 25

**Comment by 520jz** [comment]
> @circle-DING , could you please share the results?

However, I found that if the number of iterations is set at 5x10^5 of the paper, the PSNR will not achieve the effect of the paper. I trained the lego scene with four 3090s, iterated 1 million times, and the PSNR was just 20, but in the paper it was 25

**Comment by circle-DING** [comment]
I'm writing to update you on our recent training efforts. Due to limited lab resources, we utilized a 3090 Ti GPU, aiming for 5x10^5 iterations, in line with the paper's recommendation, despite the config suggesting 5x10^6.

        Upon pausing around 5x10^5 iterations, our PSNR results on data/lego/est1_events_color_bg09 were around 21, falling short of the expected outcomes mentioned in the paper. Given the time constraints and the results obtained, we've temporarily halted further training.

        I would appreciate any advice or insights on how to proceed or improve our results.
















At 2024-03-06 19:00:22, "jzyu" ***@***.***> wrote:

@circle-DING , could you please share the results?

However, I found that if the number of iterations is set at 5x10^5 of the paper, the PSNR will not achieve the effect of the paper. I trained the lego scene with four 3090s, iterated 1 million times, and the PSNR was just 20, but in the paper it was 25

**Comment by r00tman** [comment]
Hello, thank you for your question and sorry for the delay.

Could you briefly write how you compute PSNR on your side?

Just to be safe, let me briefly recap the evaluation procedure from my side.

The correct number of iterations is 5 * 10^5.

The logged PSNR should NOT be used for evaluation as it is derived from the training event-based loss of the current mini-batch. 

The numbers reported in the paper are computed after training for about 5 * 10^5 iterations using the procedure described in [metric/README.md](https://github.com/r00tman/EventNeRF/blob/main/metric/README.md).

To help the process, please also find the converged pretrained models uploaded [here](https://nextcloud.mpi-klsb.mpg.de/index.php/s/xDqwRHiWKeSRyes?path=%2F) (`models.zip`).

I hope this helps. If you have more questions, I'm available and will respond rather quickly.
